event/benches/events.rs

```diff
@@ -8,91 +8,93 @@ use std::mem::drop;
 fn criterion_benchmark(c: &mut Criterion) {
     c.bench_function("rcevent-listener-peek", move |b| {
         b.iter(|| {
-            let event = RcEventQueue::new();
+            let event = RcEventQueue::<i32>::default();
```

Comment:
I noticed that, for various types, you implemented `Default` and then forwarded `new` to `default`, any reason you don't do the same here? Further than that, why don't you let the type be inferred?

Reply:
The thing here is: In this PR, I removed several layers of indirection (e.g. wrapper structs). I could re-introduce some of them to make the 'standard API' easier to use, althrough I wasn't convinced enough that this should be done. The type isn't inferred kind-of for the same reason, because `RcEventQueue` (currently) aliases to `Rc<RefCell<RawEventQueue<T>>>` and `T` was inferred via the `emit` (was: `push`) calls, but `std::borrow::Cow` introduces an indirection at that layer, thus, the compiler can't infer the type via that way.

Reply:
(the second part about the inference might be fixed. I'll test it.)